### PR TITLE
Support for custom project module loading

### PR DESCRIPTION
**Pull request by roborourke**
This may require an addendum RFC really. It also affects how we document creating custom project only modules - I've made the PR on the documentation module reflect this approach for now.

### CODE REVIEW

**Comment by roborourke** [comment]
@rmccue ready for review / merge

File: inc/composer/class-plugin.php [hunk 1]
```diff
@@ -107,7 +107,24 @@ public function generate_module_manifest() {
...
+		$root_extra = $this->composer->getPackage()->getExtra();
+
+		if ( isset( $root_extra['altis']['modules'] ) ) {
+			foreach ( $root_extra['altis']['modules'] as $module => $config ) {
+				if ( ! isset( $config['entrypoint'] ) ) {
+					continue;
+				}
+
+				$files = array_filter( (array) $config['entrypoint'], 'file_exists' );
+				$files = implode( "';\nrequire_once dirname( __DIR__ ) . DIRECTORY_SEPARATOR . '", $files );
```
Comment (rmccue) [inline inc/composer/class-plugin.php]: Ooh, I just noticed how this is working; could we `array_map` instead, then `implode( "\n" )`? It's a little unclear as-is, and I'd definitely be confusing coming back to it.

**Review by rmccue — CHANGES_REQUESTED** [review]

File: inc/composer/class-plugin.php (hunk 1 above)
Comment (roborourke) [reply inc/composer/class-plugin.php]: Yep, you'd definitely be confusing 😬

**Comment by roborourke** [comment]
@rmccue can you re-check this one? I'll go through and merge in all the autoloader updates after this goes in.

**Review by rmccue — APPROVED** [review]

**Pull request merged**